features/FxA-89-devices-view/README.md

```diff
@@ -0,0 +1,410 @@
+
+When the user is viewing the device list
+from a web session on accounts.firefox.com,
+and is on a compatible Firefox browser
+that is not currently connected to Sync,
+we provide a "Connect..." button
+that will allow them to sign in to sync
+on that device.
+
+<img src="devices-view-current-web-session.png" alt="Zero or one Device Views" />
+
+### Disconnecting View
+
+When the user selects to disconnect a device,
+we overlay a modal confirmation dialog.
+This will minimize accidental clicks on "Disconnect..."
+and help us measure signals as to why a device is removed.
+
+In this view, we should not see
+any message to download Firefox on other devices.
+We want to minimize noise and distractions.
+
+When the user clicks and releases the "Disconnect..." button
+the following occurs:
+
+1. We verify that the user session is recent.
+   If it is too old then we bounce the user
+   through the existing "force auth" screen
+   in order to refresh their login session.
+2. We display a modal dialog
+   asking them to confirm
+   and supply a reason why they are disconnecting:
+
+   <img src="modal-disconnect-dialog.png" />
+
+3. We display an error "Valid response required"
+   if they submit the dialog
+   without selecting a reason:
+
+   <img src="modal-disconnect-dialog-validation-error.png" />
+
+4. When the user selects any option,
+   the placeholder floats up to be a field label
+   and the Disconnect button becomes enabled:
+
+   <img src="modal-disconnect-dialog-filled.png" />
+
+5. When the form is submitted
+   we disconnect the device
+   by calling the `/account/device/destroy` endpoint
+   on the server..
+
+6. Depending on the user's selected reason for disconnecting
+   we may show an additional message styled as follows:
+
+   <img src="modal-disconnect-dialog-success.png" />
+
+The available choices for disconnection reason will be:
+
+* Suspicious device
+  * Upon Disconnect, show additional message with the body copy:
+    “We’re sorry to hear about this. If this was a device you
+    really don’t trust, you should change your Firefox Account password,
+    and change any passwords saved in Firefox.”
+* Lost or stolen device
+  * Upon Disconnect, show additional message with the body copy:
+  “We’re sorry to hear about this. You should change your Firefox Account password,
+  and look for information from your device manufacturer about erasing your data remotely.”
+* Old or replaced device
+  * Upon Disconnect, don’t present an additional message.
+    Close the dialog and return to the device view.
+* Rather not say
+  * Upon Disconnect, don’t present an additional message.
+    Close the dialog and return to the device view.
+
+
+After device removal has been confirmed,
+the dialog closes and,
+the row is overtaken by the area beneath it:
+
+<img src="devices-view-disconnecting.gif" />
+
+One the server side,
+we send the user an email notification about the removed device.
```

Comment:
There are a few reasons:
-If we don't do this, the only engagement metrics around this will be "do we see people disconnecting devices or not?". If we see 25 or 50 people a day using it, what does that tell us? Nothing really. If we know why users are disconnecting, we can perhaps improve the functionality or investigate possible security issues.
-Also, if we don't know why users are disconnecting, we are unable to give accurate recommendations as per next steps to take. If a user removes a suspicious device we can give recommended next steps to regain control of their account. If we don't, they remain vulnerable. Chances are their passwords were compromised. If they lost their device, they should know they can remote wipe it. In both scenarios, the user remains vulnerable without us telling them what to do next.
-What if we display a generic message to everyone? Do we tell them to change their passwords? If we do, it might force people to reset passwords even if they didn't need to which could make for a really bad UX. If we don't tell them we leave some people vulnerable. If we give them long instructions, they won't read them and we still don't know why people are using the feature.
-Last is less functional and more of a branding reason. Firefox helps users take control of their online privacy. Through this flow, we are remaining faithful to our positioning and are helping the user take control of his privacy.

Based on this reflection, it seemed necessary to understand why people are using the feature.

If you know a way around this, I would certainly be open to recommendations. I don't want us to have to do this for nothing.